crates/core/database/src/models/server_members/ops/mongodb.rs

```diff
@@ -13,7 +15,39 @@ static COL: &str = "server_members";
 impl AbstractServerMembers for MongoDb {
     /// Insert a new server member into the database
     async fn insert_member(&self, member: &Member) -> Result<()> {
```

Comment:
For clarity, I think we should change the comment / rename the function here.
Specifically, something such as `insert_or_merge_member`, with the comment `Insert a new server member (or use the existing member if one is found)`